l10n_it_intrastat_statement/__manifest__.py

```diff
@@ -3,7 +3,7 @@
 
 {
     'name': "ITA - Dichiarazione Intrastat",
-    'version': '12.0.1.1.3',
+    'version': '12.0.1.1.4',
```

Comment:
non è più necessario aumentare la versione, viene aumentata direttamente al merge

Reply:
Grazie della review!
Sì grazie so che non è più necessario, ma penso si possa mettere lo stesso.
Dici che lo devo togliere perché crea problemi durante il merge?

Reply:
non crea nessun problema, se anche per @eLBati va bene, per me è ok.

Reply:
È bene lasciare al BOT la gestione delle versioni perché se prima di questa PR ne vengono mergiate altre, la tua andrà a portare indietro la numerazione. Il bumo di versione manuale va fatto solo se ci sono script di migrazione nella PR.

Reply:
> È bene lasciare al BOT la gestione delle versioni perché se prima di questa PR ne vengono mergiate altre, la tua andrà a portare indietro la numerazione.

In quel caso non darebbe dei conflitti durante il merge?

> Il bump di versione manuale va fatto solo se ci sono script di migrazione nella PR.

👍🏼

Reply:
No, nessun conflitto. Tu andresti semplicemente a forzare una versione del manifest che magari è già stata superata.

Reply:
Ho fatto una prova in locale e ho ottenuto un conflitto:
```
# Bump versione in un nuovo branch
(12.0 $ u=) $ git checkout -b prova_versione
Switched to a new branch 'prova_versione'
(prova_versione $) $ nano l10n_it_abicab/__manifest__.py 
(prova_versione *$) $ git add l10n_it_abicab/
(prova_versione +$) $ git commit -m "Bump version"
[prova_versione 3d5dc6373] Bump version
 1 file changed, 1 insertion(+), 1 deletion(-)
(prova_versione $) $ git show
[...]
     'name': 'Italian localization - Codici bancari ABI/CAB',
-    'version': '12.0.1.1.1',
+    'version': '12.0.2.1.1',
     'category': 'Localization/Italy',
[...]

# Bump versione in 12.0
(prova_versione $) $ git checkout 12.0
Switched to branch '12.0'
Your branch is up to date with 'origin/12.0'.
(12.0 $ u=) $ nano l10n_it_abicab/__manifest__.py 
(12.0 *$ u=) $ git add l10n_it_abicab/__manifest__.py 
(12.0 +$ u=) $ git commit -m "12.0 Bump version"
[12.0 d510e89ae] 12.0 Bump version
 1 file changed, 1 insertion(+), 1 deletion(-)
(12.0 $ u+1) $ git show
[...]     'name': 'Italian localization - Codici bancari ABI/CAB',
-    'version': '12.0.1.1.1',
+    'version': '12.0.3.0.0',
     'category': 'Localization/Italy',
[...]

# Merge
(12.0 $ u+1) $ git merge prova_versione 
Auto-merging l10n_it_abicab/__manifest__.py
CONFLICT (content): Merge conflict in l10n_it_abicab/__manifest__.py
Automatic merge failed; fix conflicts and then commit the result.
(12.0 *+$|MERGING u+1) $ cat l10n_it_abicab/__manifest__.py 
[...]
{
    'name': 'Italian localization - Codici bancari ABI/CAB',
<<<<<<< HEAD
    'version': '12.0.3.0.0',
=======
    'version': '12.0.2.1.1',
>>>>>>> prova_versione
    'category': 'Localization/Italy',
[...]
```
dici che il bot fa qualcosa di diverso? Come fa a non ottenere un conflitto?

Reply:
Se lasci gestire il bump sempre al BOT non incappi in conflitti.  Ovvio che se tu hai creato due branch ed hai modificato in entarmbi la versione ottieni un conflitto. Quello è scontato che succede.

Reply:
Ho solo riprodotto il caso che descrivevi in 
> se prima di questa PR ne vengono mergiate altre, la tua andrà a portare indietro la numerazione.

In `12.0` è stata aumentata la versione, così come in un altro branch (`prova_versione`, oppure il branch di questa PR è equivalente).
Facendo il merge di `prova_versione` in `12.0` c'è un conflitto, così come mi aspetto un conflitto mergiando questa PR se in `12.0` il modulo ha cambiato versione.

Vorrei capire com'è che funziona: io sapevo che non è necessario cambiare la versione del modulo, ma si può fare; voi mi state dicendo che non si può proprio fare?

Reply:
Nessuno sta dicendo che non si può proprio fare. Sto solo dicendo che, non avendo il pieno controllo di cosa accade da quando pubblichi la tua PR a quando viene mergiata, è preferibile lasciare al BOT in compito di gestire le versioni così prende sempre quella in essere in quel momento e la incrementa per te.

Io ti stavo solo dando un consiglio. Poi alla fine fai come preferisci.

Reply:
Grazie del consiglio :)
Se il prezzo da pagare per decidere la versione è gestire dei conflitti, in questo caso a me va bene gestirli.
Non credo ci sia il pericolo di portare indietro la versione, a meno che la PR non lo faccia esplicitamente tra le modifiche.

Reply:
Scusate se mi intrometto, ok per questa PR, non trovo indispensabile fare un commit per ripristinare la versione, ma trovo più corretto che la versione non debba essere incrementata. 
Facciamo fare al BOT quello che è suo compito, noi occupiamoci del codice :)

Reply:
Vero, il bot incrementa la versione per alleggerire il compito degli sviluppatori e, più in generale, di tutti i contributors.
Ma se uno sviluppatore masochista vuole modificarla non vedo il problema, a patto che lo faccia nel modo corretto.

Reply:
Che ti ha fatto il BOT, Simo'? Avete litigato?

Reply:
ahah sì, mi sa che ai prossimi OCA days gli rimetterò le mani addosso

Reply:
@SimoRubi con questa PR vuoi apportare modifiche a due moduli: `l10n_it_intrastat` e `l10n_it_intrastat_statement`.
Usando OCA Bot per il merge vedo solo due opzioni:

1. elimini il cambio di versione sul manifest di `l10n_it_intrastat_statement' e lo usiamo con l'opzione `patch`
2. fissi la versione anche per il modulo `l10n_it_intrastat` e lo usiamo con l'opzione 'nobump`

Io sarei per la prima per quanto già esposto precedentemente in questo 3D. A te la scelta!

Reply:
Ma il problema era questo o il rischio di _portare indietro la numerazione_? Non mi è chiaro..

Ad ogni modo, credo ci siano anche altre opzioni per cui si potrebbe mergiare così sia con `nobump` (lasciando la versione di `l10n_it_intrastat` invariata) che con `patch` (saltando una versione di `l10n_it_intrastat_statement`).
Ho aumentato anche la versione di `l10n_it_intrastat`.

Reply:
@SimoRubi Due moduli nella stessa PR non possono essere gestiti separatamente dal BOT. Il BOT gestisce tutte le numerazioni sei moduli coinvolti.

Reply:
Sì questo lo sapevo, ma pensavo si potesse comunque mergiare sia
> con `nobump` (lasciando la versione di `l10n_it_intrastat` invariata) che con `patch` (saltando una versione di `l10n_it_intrastat_statement`).

Reply:
> Sì questo lo sapevo, ma pensavo si potesse comunque mergiare sia

non ho capito come, AFAIK puoi usare solo un'opzione per l'OCA Bot. Con `nobump`, prima di https://github.com/OCA/l10n-italy/compare/95189359011d7f5cad6b5dc449aba0c2501dc9e8..af2b36c5ab7fa18cbd240d6dd530f999923515bb la versione si sarebbe aggiornata ma con una [Developmental release](https://www.python.org/dev/peps/pep-0440/#id29).

Reply:
Sì, e non sarebbe andata bene? Vengono generate continuamente, ad esempio per le traduzioni

Reply:
> Non viene mai generata una wheel con esattamente la stessa versione.

Come no? quando si mergiano le PR è così. Ciò non avviene quando vengono aggiornate le traduzioni come ad es. nel caso di https://github.com/OCA/l10n-italy/commit/db2245ba766d84735d5fbd3a7c6b4e6f45ad8f4c

Come regola generale, da ora in poi, direi di lasciare incrementare le versioni a OCA Bot, fatta eccezione nel caso in cui sono presenti script di migrazione.

Reply:
> Come no? Quando si mergiano le PR è così. Ciò non avviene quando vengono aggiornate le traduzioni come ad es. nel caso di https://github.com/OCA/l10n-italy/commit/db2245ba766d84735d5fbd3a7c6b4e6f45ad8f4c

Con `nobump` viene generata una nuova wheel con una versione `.99dev` , è scritto in https://odoo-community.org/groups/contributors-15/contributors-160247:
> - /ocabot merge nobump: will keep the version in the manifest, the drawback is that you don't have a wheel with a round version number (it will get a .dev suffix depending on the number of commits in the module since the commit that set the manifest version)


> Come regola generale, da ora in poi, direi di lasciare incrementare le versioni a OCA Bot, fatta eccezione nel caso in cui sono presenti script di migrazione.

Quali problemi ci potrebbero essere altrimenti?

Reply:
Come scrivevo in https://github.com/OCA/l10n-italy/pull/1751#discussion_r478408247:
> il bot incrementa la versione per alleggerire il compito degli sviluppatori e, più in generale, di tutti i contributors.
Ma se uno sviluppatore masochista vuole modificarla non vedo il problema, a patto che lo faccia nel modo corretto.

Quindi ora impedire di modificare la versione manualmente mi sembra fatto solo perché non si sa bene come si comporterà il bot e non mi pare giusto, così si impedirebbe di usare una funzionalità (`nobump`) che il bot ha e funziona.

Reply:
> Con `nobump` viene generata una nuova wheel con una versione `.99dev` , è scritto in https://odoo-community.org/groups/contributors-15/contributors-160247:
> 
> > * /ocabot merge nobump: will keep the version in the manifest, the drawback is that you don't have a wheel with a round version number (it will get a .dev suffix depending on the number of commits in the module since the commit that set the manifest version)

No, se lo si fissa nell'ultimo commit. Grazie per il chiarimento!

Ribadisco comunque ciò che scrivevo sopra: lascerei ad OCA Bot la gestione della versione.
Tuttavia mi confronto anche con gli altri membri del PSC in modo da avere una decisione condivisa.

Reply:
> Tuttavia mi confronto anche con gli altri membri del PSC in modo da avere una decisione condivisa.

Decisione unanime del **PSC**: lasciare sempre ad OCA Bot la gestione del cambio di versione e di ricorrere al bump manuale solo nel caso in cui siano presenti script di migrazione.

Reply:
> Quali problemi ci potrebbero essere altrimenti?

Reply:
La vera domanda a questo punto è: tu quali vantaggi hai a non usarlo?

Reply:
In realtà credo che la vera domanda sia sempre
> Quali problemi ci potrebbero essere altrimenti?

E non ho ancora trovato una risposta vera: prima mi hai scritto che *andrà a portare indietro la numerazione*, poi @tafaRU ha scritto qualcosa sul fatto che le versioni delle wheel si sovrascrivono (? questa non l'ho capita bene in realtà).
Visto che tutto il **PSC** ha deciso che dichiarare la versione in una PR è vietato (a meno di migrazioni), immagino ci possano essere problemi di qualche tipo e mi piacerebbe sapere quali possano essere perché io non ne trovo.

> tu quali vantaggi hai a non usarlo?

Ah io ho un sacco di vantaggi, non sai quanti soldi mi danno per ogni modulo dove decido la versione 😆 
Non ho nessun vantaggio particolare a farlo, mi è stato consigliato di togliere la modifica e l'ho lasciata perché pensavo di poter non seguire un consiglio.

Reply:
I problemi (come ho già detto e mi scoccia un po' ripetermi) sono dati dal fatto che se tutti modificano le versioni a mano allora abbiamo problemi di incongruenza con la conseguenza di rischiare di sballare e, eventualmente, anche di tornare indietro. Parliamo di software in cui gli aggiornamenti dovrebbero non essere un problema vista l'entità delle installazioni in ballo. Inoltre chi esegue il merge deve controllare meno cose, tra cui proprio la versione che sia coerente con quella al momento in essere. Inoltre ogni PR non deve essere ritoccata prima di andare in fase di merge e abbattiamo le possibilità di conflitti da risolvere.

Hai ammesso di non avere vantaggi a fare la modifica a mano ma ti stai ostinando a fare ostruzionismo per una palese presa di posizione che non capisco che finalità abbia.

Con questa chiudo perché, senza offesa, parlare di numeretti di moduli non è la più accattivante delle attività.

Reply:
@tafaRU puoi aggiungere https://github.com/OCA/l10n-italy/pull/1751#discussion_r488655063 nelle linee guida italiane per lo sviluppo (https://www.odoo-italia.org/documentazione/12.0/sviluppo.html)?
Così anche gli altri contributors sanno che non si può fare e abbiamo un punto di riferimento, invece di cercare sempre il link al commento come hai fatto in https://github.com/OCA/l10n-italy/pull/1871#issuecomment-693253477